src/frontend/src/optimizer/plan_node/logical_over_window.rs

```diff
@@ -263,6 +263,57 @@ impl LogicalOverWindow {
             .collect();
         Self::new(window_functions, input)
     }
+
+    pub fn split_with_rule(&self, group_rule: Vec<usize>) -> PlanRef {
+        assert!(group_rule.len() == self.window_functions().len());
+
+        let mut output_proj_builder = ProjectBuilder::default();
```

Comment:
I think we should not use the ProjectBuilder here, because we do not want to deduplicate the same expression, or we can not make sure the output schema is the same with the original plan